tests/cli/main.rs

```diff
@@ -274,6 +274,119 @@ fn basic_backup() {
     // TODO: Compare vs source tree.
 }
```

Comment:
There are also some tests for "old formats" to check we can read archives written by old versions, from `testdata/archive`. When this is finalized we should check in a small archive there.

It would also be good to check `ls -l` on an old archive. I think it will just work because it's stored in Options.

Reply:
The mode and owner should be set to the default if they aren't found in the archive. Currently I think that means the owner would be "", which doesn't make sense. I think it should probably be set to the user running conserve if it can't be found in the archive, since that's the user that will own the files upon restoration.

Reply:
I made a test `long_listing_old_archive` that calls `show::show_entry_names` on the `MINIMAL_ARCHIVE_VERSIONS`. Metadata just shows up as `rwxrwxr-x none none` because with no data stored it just assumes the default values. 
An improvement would be to set the default metadata more intelligently (based on current user/group and inode type) but I don't think it's worth the trouble.

Reply:
That seems OK for listing. It might be nicer to show the permissions also as `none` though.

It does make me think that we should test that _restore_ from an archive that has no permissions or owners should not chmod or chown the files.

Reply:
We could have the unix_mode be stored as an `Option<u32>` so that the default is `None`. Testing that it doesn't chown or chmod the files would be tricky, since the files typically don't exist prior to restoring them from the archive. Could be done by invoking API functions, but I think if the mode and owner are stored as `None` that's reasonable enough assurance that nothing will be overridden.